etp-backend/src/main/clj/solita/etp/service/aineisto.clj

```diff
@@ -65,3 +67,50 @@
       (doseq [aineisto aineistot]
         (set-access! db kayttaja-id aineisto))
       1)))
+
+(defn extract-byte-array-and-reset! [buf]
```

Comment:
Tässä on varmaan aina tiedossa mitä luokkaa tuo buf on? Voisi näin Java-interopia käytettäessä laittaa tälle parametrille type hintin siitä luokasta, niin sitten editoritkin osaa näyttää noiden Java-metodien kutsuista lisätietoa. Sinällään toteutukseen sillä ei ole tällä hetkellä vaikutusta, koska reflektiosta ei tule varoitusta (reflektion kanssa voisi olla heikompi suorituskyky verrattuna siihen, että type hintistä voidaan suoraan päätellä käytettävä luokka), mutta näin kehittäjäkokemuksen ja ymmärrettävyyden kannalta voisi olla parempi lisätä tähän tuo type hint.